### PR TITLE
Update remarks for empty search strings

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to dotnet/docs#21372.

Removes any references to return values when searching for empty or zero-weight strings, since this behavior is nuanced.

The only changes to the snippet files are to reformat them and to remove examples that search for zero-weight strings, since the results can vary.

*Hide whitespace changes in diff.*

### CODE REVIEW

**Comment by gewarren** [comment]
Also need to update: CompareInfo.LastIndexOf (and possibly MemoryExtensions.LastIndexOf).

File: samples/snippets/csharp/VS_Snippets_CLR_System/system.String.LastIndexOf/cs/lastindexof22.cs [hunk 1]
```diff
@@ -1,59 +1,46 @@
...
-         Console.WriteLine(s2.LastIndexOf("\u00AD", position));
-      
-      // Find the index of the soft hyphen followed by "n".
-      position = s1.LastIndexOf("m");
-      Console.WriteLine("'m' at position {0}", position);
-      if (position >= 0)
-         Console.WriteLine(s1.LastIndexOf("\u00ADn", position));
+        // Find the index of the soft hyphen followed by "n".
+        position = s1.LastIndexOf("m");
+        Console.WriteLine("'m' at position {0}", position);
```
Comment (BillWagner) [inline samples/snippets/csharp/VS_Snippets_CLR_System/system.String.LastIndexOf/cs/lastindexof22.cs:14]: Consider interpolated strings here:

```suggestion
        Console.WriteLine($"'m' at position {position}");

```

File: samples/snippets/csharp/VS_Snippets_CLR_System/system.String.LastIndexOf/cs/lastindexof22.cs [hunk 2]
```diff
@@ -1,59 +1,46 @@
...
+            Console.WriteLine(s2.LastIndexOf("\u00ADn", position));
 
-      position = s2.LastIndexOf("m");
-      Console.WriteLine("'m' at position {0}", position);
-      if (position >= 0)   
-         Console.WriteLine(s2.LastIndexOf("\u00ADm", position));
-   }
+        // Find the index of the soft hyphen followed by "m".
+        position = s1.LastIndexOf("m");
+        Console.WriteLine("'m' at position {0}", position);
```
Comment (BillWagner) [inline samples/snippets/csharp/VS_Snippets_CLR_System/system.String.LastIndexOf/cs/lastindexof22.cs:25]: same:

```suggestion
        Console.WriteLine($"'m' at position {position}");

```

File: samples/snippets/csharp/VS_Snippets_CLR_System/system.String.LastIndexOf/cs/lastindexof22.cs [hunk 3]
```diff
@@ -1,59 +1,46 @@
...
+        if (position >= 0)
+            Console.WriteLine(s1.LastIndexOf("\u00ADm", position));
+
+        position = s2.LastIndexOf("m");
+        Console.WriteLine("'m' at position {0}", position);
```
Comment (BillWagner) [inline samples/snippets/csharp/VS_Snippets_CLR_System/system.String.LastIndexOf/cs/lastindexof22.cs:30]: same:

```suggestion
        Console.WriteLine($"'m' at position {position}");

```

File: samples/snippets/csharp/VS_Snippets_CLR_System/system.String.LastIndexOf/cs/lastindexof23.cs [hunk 4]
```diff
@@ -1,59 +1,46 @@
...
-         Console.WriteLine(s2.LastIndexOf("\u00AD", position, position + 1));
-      
-      // Find the index of the soft hyphen followed by "n".
-      position = s1.LastIndexOf("m");
-      Console.WriteLine("'m' at position {0}", position);
-      if (position >= 0)
-         Console.WriteLine(s1.LastIndexOf("\u00ADn", position, position + 1));
+        // Find the index of the soft hyphen followed by "n".
+        position = s1.LastIndexOf("m");
+        Console.WriteLine("'m' at position {0}", position);
```
Comment (BillWagner) [inline samples/snippets/csharp/VS_Snippets_CLR_System/system.String.LastIndexOf/cs/lastindexof23.cs:14]:
```suggestion
        Console.WriteLine($"'m' at position {position}");

```

File: samples/snippets/csharp/VS_Snippets_CLR_System/system.String.LastIndexOf/cs/lastindexof23.cs [hunk 5]
```diff
@@ -1,59 +1,46 @@
...
-      if (position >= 0)   
-         Console.WriteLine(s2.LastIndexOf("\u00ADn", position, position + 1));
-      
-      // Find the index of the soft hyphen followed by "m".
-      position = s1.LastIndexOf("m");
-      Console.WriteLine("'m' at position {0}", position);
-      if (position >= 0)
-         Console.WriteLine(s1.LastIndexOf("\u00ADm", position, position + 1));
+        position = s2.LastIndexOf("m");
+        Console.WriteLine("'m' at position {0}", position);
```
Comment (BillWagner) [inline samples/snippets/csharp/VS_Snippets_CLR_System/system.String.LastIndexOf/cs/lastindexof23.cs:19]:
```suggestion
        Console.WriteLine($"'m' at position {position}");

```

File: samples/snippets/csharp/VS_Snippets_CLR_System/system.String.LastIndexOf/cs/lastindexof23.cs [hunk 6]
```diff
@@ -1,59 +1,46 @@
...
+            Console.WriteLine(s2.LastIndexOf("\u00ADn", position, position + 1));
 
-      position = s2.LastIndexOf("m");
-      Console.WriteLine("'m' at position {0}", position);
-      if (position >= 0)   
-         Console.WriteLine(s2.LastIndexOf("\u00ADm", position, position + 1));
-   }
+        // Find the index of the soft hyphen followed by "m".
+        position = s1.LastIndexOf("m");
+        Console.WriteLine("'m' at position {0}", position);
```
Comment (BillWagner) [inline samples/snippets/csharp/VS_Snippets_CLR_System/system.String.LastIndexOf/cs/lastindexof23.cs:25]:
```suggestion
        Console.WriteLine($"'m' at position {position}");

```

File: samples/snippets/csharp/VS_Snippets_CLR_System/system.String.LastIndexOf/cs/lastindexof23.cs [hunk 7]
```diff
@@ -1,59 +1,46 @@
...
+        if (position >= 0)
+            Console.WriteLine(s1.LastIndexOf("\u00ADm", position, position + 1));
+
+        position = s2.LastIndexOf("m");
+        Console.WriteLine("'m' at position {0}", position);
```
Comment (BillWagner) [inline samples/snippets/csharp/VS_Snippets_CLR_System/system.String.LastIndexOf/cs/lastindexof23.cs:30]:
```suggestion
        Console.WriteLine($"'m' at position {position}");

```

File: samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.LastIndexOf/vb/lastindexof22.vb [hunk 8]
```diff
@@ -1,66 +1,51 @@
...
+    Public Sub Main()
+        ' <Snippet22>
+        Dim position As Integer
+        Dim softHyphen As String = ChrW(&HAD)
+        Dim s1 As String = "ani" + softHyphen + "mal"
+        Dim s2 As String = "animal"
+
+        ' Find the index of the soft hyphen followed by "n".
+        position = s1.LastIndexOf("m")
+        Console.WriteLine("'m' at position {0}", position)
```
Comment (BillWagner) [inline samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.LastIndexOf/vb/lastindexof22.vb:14]: VB as well (several occurrences):

```suggestion
        Console.WriteLine($"'m' at position {position}")

```

**Review by BillWagner — APPROVED** [review]
This looks great @gewarren 

I made suggestions to use interpolated strings instead of positional formatting in a number of places. After those changes, you can :shipit:

File: samples/snippets/csharp/VS_Snippets_CLR_System/system.String.LastIndexOf/cs/lastindexof22.cs [hunk 9]
```diff
@@ -1,59 +1,46 @@
...
-      if (position >= 0)   
-         Console.WriteLine(s2.LastIndexOf("\u00ADn", position));
-      
-      // Find the index of the soft hyphen followed by "m".
-      position = s1.LastIndexOf("m");
-      Console.WriteLine("'m' at position {0}", position);
-      if (position >= 0)
-         Console.WriteLine(s1.LastIndexOf("\u00ADm", position));
+        position = s2.LastIndexOf("m");
+        Console.WriteLine("'m' at position {0}", position);
```
Comment (gewarren) [inline samples/snippets/csharp/VS_Snippets_CLR_System/system.String.LastIndexOf/cs/lastindexof22.cs:19]:
```suggestion
        Console.WriteLine($"'m' at position {position}");

```

File: samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.LastIndexOf/vb/lastindexof22.vb [hunk 10]
```diff
@@ -1,66 +1,51 @@
...
+        If position >= 0 Then
+            Console.WriteLine(s1.LastIndexOf(softHyphen + "n", position))
+        End If
+
+        position = s2.LastIndexOf("m")
+        Console.WriteLine("'m' at position {0}", position)
```
Comment (gewarren) [inline samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.LastIndexOf/vb/lastindexof22.vb:20]:
```suggestion
        Console.WriteLine($"'m' at position {position}")

```

File: samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.LastIndexOf/vb/lastindexof22.vb [hunk 11]
```diff
@@ -1,66 +1,51 @@
...
+        If position >= 0 Then
+            Console.WriteLine(s2.LastIndexOf(softHyphen + "n", position))
+        End If
+
+        ' Find the index of the soft hyphen followed by "m".
+        position = s1.LastIndexOf("m")
+        Console.WriteLine("'m' at position {0}", position)
```
Comment (gewarren) [inline samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.LastIndexOf/vb/lastindexof22.vb:27]:
```suggestion
        Console.WriteLine($"'m' at position {position}")

```

File: samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.LastIndexOf/vb/lastindexof22.vb [hunk 12]
```diff
@@ -1,66 +1,51 @@
...
+        If position >= 0 Then
+            Console.WriteLine(s1.LastIndexOf(softHyphen + "m", position))
+        End If
+
+        position = s2.LastIndexOf("m")
+        Console.WriteLine("'m' at position {0}", position)
```
Comment (gewarren) [inline samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.LastIndexOf/vb/lastindexof22.vb:33]:
```suggestion
        Console.WriteLine($"'m' at position {position}")

```

File: samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.LastIndexOf/vb/lastindexof23.vb [hunk 13]
```diff
@@ -1,67 +1,51 @@
...
+        ' <Snippet23>
+        Dim position As Integer
+        Dim softHyphen As String = ChrW(&HAD)
+
+        Dim s1 As String = "ani" + softHyphen + "mal"
+        Dim s2 As String = "animal"
+
+        ' Find the index of the soft hyphen followed by "n".
+        position = s1.LastIndexOf("m")
+        Console.WriteLine("'m' at position {0}", position)
```
Comment (gewarren) [inline samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.LastIndexOf/vb/lastindexof23.vb:15]:
```suggestion
        Console.WriteLine($"'m' at position {position}")

```

File: samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.LastIndexOf/vb/lastindexof23.vb [hunk 14]
```diff
@@ -1,67 +1,51 @@
...
+        If position >= 0 Then
+            Console.WriteLine(s1.LastIndexOf(softHyphen + "n", position, position + 1))
+        End If
+
+        position = s2.LastIndexOf("m")
+        Console.WriteLine("'m' at position {0}", position)
```
Comment (gewarren) [inline samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.LastIndexOf/vb/lastindexof23.vb:21]:
```suggestion
        Console.WriteLine($"'m' at position {position}")

```

File: samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.LastIndexOf/vb/lastindexof23.vb [hunk 15]
```diff
@@ -1,67 +1,51 @@
...
+        If position >= 0 Then
+            Console.WriteLine(s2.LastIndexOf(softHyphen + "n", position, position + 1))
+        End If
 
+        ' Find the index of the soft hyphen followed by "m".
+        position = s1.LastIndexOf("m")
+        Console.WriteLine("'m' at position {0}", position)
```
Comment (gewarren) [inline samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.LastIndexOf/vb/lastindexof23.vb:28]:
```suggestion
        Console.WriteLine($"'m' at position {position}")

```

File: samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.LastIndexOf/vb/lastindexof23.vb [hunk 16]
```diff
@@ -1,67 +1,51 @@
...
+        If position >= 0 Then
+            Console.WriteLine(s1.LastIndexOf(softHyphen + "m", position, position + 1))
+        End If
+
+        position = s2.LastIndexOf("m")
+        Console.WriteLine("'m' at position {0}", position)
```
Comment (gewarren) [inline samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.LastIndexOf/vb/lastindexof23.vb:34]:
```suggestion
        Console.WriteLine($"'m' at position {position}")

```

**Comment by gewarren** [comment]
Thanks for reviewing this, Bill!

**Pull request merged**